src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -33,27 +35,67 @@ public GenericTypeParameter(string name, IPythonModule declaringModule, IReadOnl
         public override PythonMemberType MemberType => PythonMemberType.Generic;
         public override bool IsSpecialized => true;
 
-
-        public static IPythonType FromTypeVar(IArgumentSet argSet, IPythonModule declaringModule, IndexSpan location = default) {
+        private static bool TypeVarArgumentsValid(IArgumentSet argSet) {
             var args = argSet.Values<IMember>();
+            var eval = argSet.Eval;
+            var callExpression = argSet.CallExpression;
+            var callLocation = callExpression?.GetLocation(eval.Module);
+
             if (args.Count == 0) {
-                // TODO: report that at least one argument is required.
-                return declaringModule.Interpreter.UnknownType;
+                eval.ReportDiagnostics(
+                    eval.Module?.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarMissingFirstArgument,
+                    callLocation?.Span ?? default,
+                    Diagnostics.ErrorCodes.TypeVarArguments,
+                    Severity.Error, DiagnosticSource.Analysis)
+                );
+
+                return false;
             }
 
             var name = (args[0] as IPythonConstant)?.GetString();
             if (string.IsNullOrEmpty(name)) {
-                // TODO: report that type name is not a string.
-                return declaringModule.Interpreter.UnknownType;
+                var firstArgLocation = callExpression?.Args[0]?.GetLocation(eval.Module);
+                eval.ReportDiagnostics(
+                    eval.Module?.Uri,
+                    new DiagnosticsEntry(Resources.TypeVarFirstArgumentNotString,
+                    firstArgLocation?.Span ?? default,
+                    Diagnostics.ErrorCodes.TypeVarArguments,
+                    Severity.Warning, DiagnosticSource.Analysis)
+                );
+
+                return false;
+            }
+
+            if (args.Count - 1 == 1) {
```

Comment:
Why not just `args.Count == 0`?

Reply:
Err, I mean `== 2`.

Reply:
So when calling TypeVar, we have something like T = TypeVar('T', int). So the length of args is 2. Python gives a runtime error if there is only one constraint, (int in this case), so we exclude the first initial 'T' from our count, which is why we subtract 1.

Reply:
I know, I just mean from a style point of view, checking `x - 1 == 1` versus `x == 2`.

Reply:
Yup didn't see your earlier message, I changed it.